crates/driver/src/commit_reveal.rs

```diff
@@ -39,23 +45,272 @@ pub trait CommitRevealSolving: Send + Sync {
 // For now this wrapper is only a compatibility layer to let us use the new driver with existing
 // solvers for faster development.
 pub struct CommitRevealSolver {
-    #[allow(dead_code)]
-    solver: Box<dyn Solver>,
+    solver: Arc<dyn Solver>,
+    settlement_rater: Arc<dyn SettlementRating>,
+    gas_estimator: Arc<dyn GasPriceEstimating>,
+    stored_solution: Mutex<Option<(SettlementSummary, Settlement)>>,
 }
 
 impl CommitRevealSolver {
-    pub fn new(solver: Box<dyn Solver>) -> Self {
-        Self { solver }
+    pub fn new(
+        solver: Arc<dyn Solver>,
+        settlement_rater: Arc<dyn SettlementRating>,
+        gas_estimator: Arc<dyn GasPriceEstimating>,
+    ) -> Self {
+        Self {
+            solver,
+            settlement_rater,
+            gas_estimator,
+            stored_solution: Mutex::new(Default::default()),
+        }
+    }
+
+    async fn _commit(&self, auction: Auction) -> Result<(SettlementSummary, Settlement)> {
```

Comment:
Nit: Starting names with an underscore in Rust has special meaning which makes the compiler ignore this item for the "unused variable/type" warning. Since that isn't the reason for the underscore here I'd put it at the end `commit_`.

Reply:
Renamed it `commit_impl()` in the hopes of being slightly more informative.